### PR TITLE
Add blogpost "Integrate Angular in Spring Boot using Gradle"

### PR DESCRIPTION
**Pull request by gurtjun**

### CODE REVIEW

File: _posts/2018-10-12-spring-boot-angular-gradle.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+image: /img/2018-10-12-spring-boot-angular-gradle/angular-spring-boot-gradle.jpg
+tags: [Angular, Spring Boot, Gradle]
+category: Architecture
+comments: true
+---
+
+I often found myself struggling when I had to integrate Angular into a Spring Boot project. This is something that should
+not take a lot of your valuable development time. That's why I want to share how I did it with Gradle in a very fast and easy way.
+
+I setup an example repo which you can find [on GitHub](https://github.com/gurtjun/angular-spring-boot-gradle){:target="_blank"}.
```
Comment (yannickdeturck) [inline _posts/2018-10-12-spring-boot-angular-gradle.md]: set up

File: _posts/2018-10-12-spring-boot-angular-gradle.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+## Application structure
+
+This guide assumes you have a root directory that contains two child directories. 
+One with the Angular code, and another one with the Spring Boot code. By keeping these apart from each other
+it will be easier to develop within the application. 
+
+We'll make use of [Gradle's multi-project builds](https://docs.gradle.org/current/userguide/intro_multi_project_builds.html)
```
Comment (yannickdeturck) [inline _posts/2018-10-12-spring-boot-angular-gradle.md]: Also add `{:target="_blank" rel="noopener noreferrer"}` to the link.

File: _posts/2018-10-12-spring-boot-angular-gradle.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+to split the application into multiple modules. 
+
+Because I generated my Spring Boot project with [Spring Initializr](https://start.spring.io) I already have a
```
Comment (yannickdeturck) [inline _posts/2018-10-12-spring-boot-angular-gradle.md]: Also add `{:target="_blank" rel="noopener noreferrer"}` to the link.

File: _posts/2018-10-12-spring-boot-angular-gradle.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+Gradle will now recognize both child directories as a subproject. 
+
+## Angular
+
+For the Angular part we want create a jar with a `static` directory that contains the result of our Angular build. 
+By doing this we can include the jar in our backend module. And because Spring Boot will automatically add static web 
+resources located within `static`, the Angular application will be visible when we launch the application.
+
+This can be done by using the [com.moowork.node](https://plugins.gradle.org/plugin/com.moowork.node) plugin.
```
Comment (yannickdeturck) [inline _posts/2018-10-12-spring-boot-angular-gradle.md]: Also add `{:target="_blank" rel="noopener noreferrer"}` to the link.

File: _posts/2018-10-12-spring-boot-angular-gradle.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+Our Spring Boot project already has a `build.gradle` file generated. 
+We just have to add one line within our dependencies to include the todo-ui module.
+
+```groovy
+dependencies {
+	implementation(project(':todo-ui'))
+}
+```
+
+Build the project, execute the generated Spring Boot jar and go to [localhost:8080](),you should see your Angular web application.
```
Comment (yannickdeturck) [inline _posts/2018-10-12-spring-boot-angular-gradle.md]: Also add `{:target="_blank" rel="noopener noreferrer"}` to the link (even though no one will probably end up clicking it :)).

Comment (yannickdeturck) [reply _posts/2018-10-12-spring-boot-angular-gradle.md]: Also add a space after the second comma. 🤓

File: _posts/2018-10-12-spring-boot-angular-gradle.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+## Note
+
+There are several ways to integrate Angular in Spring Boot using Gradle (or Maven). 
+To me, this is the easiest and fastest way to do it. Because we've separated the backend from the frontend it will be 
+a lot easier to find your way in both modules.
+
+Now go and apply this on your own projects! Don't hesitate to contact me if you have any questions.
+
+<sub><sup>
+<sup>1</sup> by default, `npm run build` will execute `ng build` (specified in `package.json`).<br/>
```
Comment (yannickdeturck) [inline _posts/2018-10-12-spring-boot-angular-gradle.md]: Can omit the period at the end in order to be consistent with the second sup.

**Comment by yannickdeturck** [comment]
LGTM! Leaving it open for just a bit to see if others want to check it out as well seeing as the publication date has been set to tomorrow.

File: _posts/2018-10-12-spring-boot-angular-gradle.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+``` 
+
+We'll use `todo` as the project name and include both the backend and frontend module by specifying their directory name.
+It's important that this name matches the path of the directory, otherwise Gradle cannot find these modules.
+
+Gradle will now recognize both child directories as a subproject. 
+
+## Angular
+
+For the Angular part we want create a jar with a `static` directory that contains the result of our Angular build.
```
Comment (Deryaa) [inline _posts/2018-10-12-spring-boot-angular-gradle.md]: want to* create

Comment (gurtjun) [reply _posts/2018-10-12-spring-boot-angular-gradle.md]: Good find, @Deryaa!

**Pull request merged**